### PR TITLE
Uses new frictionless report component to display validation report

### PR DESCRIPTION
**Pull request by aivuk**
Uses React Frictionless [Report](https://components.frictionlessdata.io/) component to display a resource validation results.

### CODE REVIEW

**Comment by amercader** [comment]
@aivuk This is looking good, but there are a couple of things to iron out:

* When you click on the report page:
    * The badge shows "error" when it should show "invalid"
    * The duration of the check is not populated
    * We are showing the local path in the report, which we shouldn't. IIRC this was corrected [here](https://github.com/frictionlessdata/ckanext-validation/blob/455090bd8fd27fbf0d560850e72df5a7e067b6f6/ckanext/validation/jobs.py#L92) but it might need some adjusting to the new format
Screenshot 2022-10-21 at 12-51-50 Validation Report - CKAN

* When you work in sync mode (`ckanext.validation.run_on_create_sync = True` / `ckanext.validation.run_on_update_sync = True`) the files are validated in the same request where you create the resource. If there are errors, the validation report is returned. We hijack the error message in the form to not display the raw JSON report but a link instead, that opens a dialog with the report component. The component is not currently rendered in the dialog:

Screenshot 2022-10-21 at 13-04-53 Add data to the dataset - CKAN
Screenshot 2022-10-21 at 13-09-06 Add resource - Test validation 2 - CKAN

The [errors.html](https://github.com/frictionlessdata/ckanext-validation/blob/dev-v2/ckanext/validation/templates/scheming/snippets/errors.html) template is a good starting point to investigate

**Comment by amercader** [comment]
Sorry, just realized this was a draft, so apologies if you were already looking at these

**Comment by aivuk** [comment]
> Sorry, just realized this was a draft, so apologies if you were already looking at these

Thanks, I was working on the dialog, but I'm having an error using the report.js. For some reason the validation_error json is not being passed to the module-validation-report.js (it's undefined), but I checked and the json is there in the template validation/snippets/validation_report_dialog.html but it's not being passed to the JS module. I'm investigating it. I missed the error in the badge, I thought that it would need to be "error", but I got it that we need to have three states.

undefined-report-json

**Comment by aivuk** [comment]
Hi @amercader, the following points that you raised are fixed:

- [x] The badge shows "error" when it should show "invalid"
- [x] The duration of the check is not populated
- [x] We are showing the local path in the report, which we shouldn't. IIRC this was corrected here but it might need some adjusting to the new format

I'm still debugging why when the plugin is running in sync mode, the report is not being displayed by the component. The same thing happens when the plugin is not running in sync, after you update a resource and click in the badge label before reloading the page.

**Comment by amercader** [comment]
Thanks @aivuk , that sounds great. If it helps this JSON.parse() seems to be failing [here](https://github.com/frictionlessdata/ckanext-validation/blob/use-frictionless-report-component/ckanext/validation/webassets/js/module-validation-report.js#L10) but I'm not sure if that's related

File: README.md [hunk 1]
```diff
@@ -72,9 +72,9 @@ Create the database tables running:
 
 ## Configuration
 
-Once installed, add the `validation` plugin to the `ckan.plugins` configuration option in your INI file:
+Once installed, add the `validation` plugin to the `ckan.plugins` configuration option in your INI file. If using ckanext-scheming, the `validation` plugins should be loaded **before** the `scheming_datasets` one:
```
Comment (ThrawnCA) [inline README.md:75]: Is it feasible to remove this constraint? I know that plugin ordering can be important, but perhaps there is a way to make them more independent?

**Comment by ThrawnCA** [comment]
Is it necessary to copy the Frictionless web assets into this repo? Are they not available in any PyPI package?

**Comment by aivuk** [comment]
> Is it necessary to copy the Frictionless web assets into this repo? Are they not available in any PyPI package?

The frictionless web assets are from https://components.frictionlessdata.io/ and the are a JS package living on npm https://www.npmjs.com/package/frictionless-components. I'm using the latest build (version 1.3.2) instead of a CDN or adding a build it step in ckanext-validation that would download and build the npm package.

File: README.md (hunk 1 above)
Comment (amercader) [reply README.md:75]: @ThrawnCA I don't think so. We want ckanext-validation's [errors.html](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/templates/scheming/snippets/errors.html) to override ckanext-scheming's so we can hook the report to the error messages.

I guess that we could add "If using ckanext-scheming and using synchronous mode, the `validation` plugin should be loaded **before** the `scheming_datasets` one" but that would make the instructions a bit more confusing

Comment (ThrawnCA) [reply README.md:75]: Oh, and ckanext-scheming defines the base version of that template, which doesn't extend anything, so if ckanext-scheming comes first, then processing stops and ckanext-validation's version is never reached. That makes sense.

**Comment by aivuk** [comment]
@amercader Fixed the report in sync mode, now the report is being displayed in a dialog if there are errors when creating a new resource.

**Comment by amercader** [comment]
Great stuff! Can you make the dialog a bit wider? it's a bit hard to read currently. This probably helps: https://getbootstrap.com/docs/4.2/components/modal/#optional-sizes

Screenshot 2022-10-26 at 15-38-47 Add resource - Test validation 2 - CKAN

Besides this, this is good to merge once #69 is done

File: ckanext/validation/webassets/js/module-modal-dialog.js [hunk 2]
```diff
@@ -15,6 +17,7 @@ this.ckan.module('modal-dialog', function (jQuery) {
      */
     initialize: function () {
       jQuery.proxyAll(this, /_on/);
+      console.log('HERE', this)
```
Comment (ThrawnCA) [inline ckanext/validation/webassets/js/module-modal-dialog.js:20]: What is the purpose of this log?

Comment (aivuk) [reply ckanext/validation/webassets/js/module-modal-dialog.js:20]: thanks for pointing it, I already removed it. It was there to debug why the report json was working in the validation report template but it wasn't in the modal dialog.

**Pull request merged**